templates/web/smidsy/reports/index.html

```diff
@@ -1,55 +1,142 @@
-  Show:
-  <label><input type="radio" name="type" value="all" 
-      [% IF report_filter == 'all' %]checked[% END %]> 
-      All bodies</label>
-  <label><input type="radio" name="type" value="london"
-      [% IF report_filter == 'london' %]checked[% END %]> 
-      London boroughs</label>
-  <label><input type="radio" name="type" value="city"
-      [% IF report_filter == 'city' %]checked[% END %]> 
-      Cities (unitary authorities)</label>
-  <label><input type="radio" name="type" value="dc"
-      [% IF report_filter == 'dc' %]checked[% END %]> 
-      Districts and counties</label>
+<form method="GET">
+    <div class="filters">
+        <p>
+            [% IF area_body %]
+            <input type="hidden" name="area_body" value="[% area_body %]">
+            [% END %]
+            <label for="area">[% loc('Incidents reported in:') %]</label>
+            <select class="form-control" id="area" name="area">
+                <option value="">All areas</option>
+                [% IF area_body %]
+                <option value="wards"[% ' selected' IF area == area_body OR area == 'wards' %]>All wards</option>
+                [% FOR c IN children.sort('name') %]
+                <option value="[% c.id %]"[% ' selected' IF c.id == area %]>[% c.name | html ~%]</option>
+                [% END %]
+                [% ELSE %]
+                [% FOR b IN bodies # Not body as 'body' may be on stash %]
+                <option value="[% b.id %]"[% ' selected' IF b.id == area %]>[% b.name | html ~%]
+                    [% IF NOT b.get_column("area_count") %] [% loc('(no longer exists)') %][% END ~%]
+                </option>
+                [% END %]
+                [% END %]
+            </select>
+        </p>
+        <p>
+            <label for="start_date">[% loc('Start date:') %]</label>
+            <input type="date" class="form-control" id="start_date" name="start_date" value="[% date.format(start_date, '%Y-%m-%d', 'en_GB') %]">
+        </p>
+        <p>
+            <label for="end_date">[% loc('End date:') %]</label>
+            <input type="date" class="form-control" id="end_date" name="end_date" value="[% date.format(end_date, '%Y-%m-%d', 'en_GB') %]">
+        </p>
+        <p>
+            <label for="sources">[% loc('Data sources') %]</label>
+            <select class="form-control" id="sources" name="sources">
+                <option value="">Collideoscope &amp; STATS19</option>
+                <option value="collideoscope"[% ' selected' IF sources == 'collideoscope' %]>Collideoscope only</option>
+                <option value="stats19"[% ' selected' IF sources == 'stats19' %]>STATS19 only</option>
+            </select>
+        </p>
+        <p>
+            <label for="transport">[% loc('Transport') %]</label>
+            <select class="form-control" id="transport" name="transport">
+                <option value="">All</option>
+                [% FOR p IN participant_labels.keys %]
```

Comment:
Perhaps use `pairs` here (and then `.key` and `.value` below) so it's sorted? Plus tiny bit clearer.